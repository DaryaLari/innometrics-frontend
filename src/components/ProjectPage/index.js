import React from "react";
import { withRouter } from 'react-router'
import { Link } from 'react-router-dom'
import styles from "./style.css";

class ProjectPage extends React.Component {
  render() {
    console.log(this.props)
    const projectName = this.props.match.params.projectName
    return (
      <div className={styles.content}>
        <h1><Link to='/projects'>Projects</Link> / {projectName}</h1>
        <div className={styles.statistics}>
          Statistics in diagrams would be here
        </div>
      </div>

    )
  }
}

export default withRouter(ProjectPage);