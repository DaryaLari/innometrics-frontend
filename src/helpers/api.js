import axios from "axios";

export const DOMAIN_ADDRESS = 'http://188.130.155.81:8120'

const request = axios.create({
  baseURL: DOMAIN_ADDRESS,
  headers: {
    'Content-Type': 'multipart/form-data'
  }
})

export function postRequest (url, params={}, withToken=false) {

  const formData = new FormData()
  Object.keys(params).forEach(key => formData.append(key, params[key]))

  return new Promise((resolve, reject) => {
    request.post(
      url,
      formData,
      {
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(error.response)
        })
  })
}

export function getRequest (url, params={}, withToken=false) {

  return new Promise((resolve, reject) => {
    request.get(
      url,
      {
        params: params,
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(error.response)
        })
  })
}

const additionalHeaders = (isWithToken) => isWithToken ? {'X-API-Key': getToken()} : {}
