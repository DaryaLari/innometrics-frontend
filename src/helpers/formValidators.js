import _ from 'lodash'

export const required = (value) => {
  return !_.isEmpty(_.trim(value)) ?
      undefined : 'Required'
}

export const email = (value) => {
  return /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(_.trim(value)) ?
      undefined : 'Invalid email'
}