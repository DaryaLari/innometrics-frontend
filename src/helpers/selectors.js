import {createSelector} from "reselect";

export const userAuthorized = createSelector(
  (state) => {
    return state.user
  },
  (user) => {
    return user.authorized
  }
)
