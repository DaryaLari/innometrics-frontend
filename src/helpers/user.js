export const isTokenExists = () => {
  return null != localStorage.getItem("user")
      && null != JSON.parse(localStorage.getItem("user")).token
}

export const getToken = () => {
  if(isTokenExists()){
    return JSON.parse(localStorage.getItem("user")).token
  }
  return ''
}

export const saveUserToLocalStorage = (token) => {
  localStorage.setItem("user", JSON.stringify({token: token}))
}

export const removeUserFromLocalStorage = () => {
  localStorage.removeItem("user")
}