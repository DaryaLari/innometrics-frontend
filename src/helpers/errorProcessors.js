import {store} from '../store'
import { TYPES as USER_TYPES } from '../store/user/actionTypes'

export const processApiError = (error) => {

  switch(_.get(error.response, 'status', 0)){
    case 0:
      return {status: 0, statusText: 'No Internet connection', data: { message: 'No Internet connection'}}
    case 401:
      store.dispatch({type: USER_TYPES.UNAUTHORIZED})
      return error.response
    default:
      return error.response
  }
}