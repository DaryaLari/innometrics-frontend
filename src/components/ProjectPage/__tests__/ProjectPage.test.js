import React from 'react'
import ProjectPage from '../index'
import { shallow } from 'enzyme'
import { history } from '../../../helpers/history'
import { Router } from 'react-router-dom'

describe('Test ProjectPage component', () => {

  it('Test shallow render ProjectPage component', () => {
    const wrapper = shallow(
      <Router history={history}>
        <ProjectPage />
      </Router>
    )

    expect(wrapper).toMatchSnapshot()
  })

})