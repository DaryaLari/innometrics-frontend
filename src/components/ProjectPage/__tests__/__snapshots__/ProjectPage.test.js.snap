// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectPage component Test shallow render ProjectPage component 1`] = `<withRouter(ProjectPage) />`;
